src/libcollections/btree/map.rs

```diff
@@ -1867,6 +1886,17 @@ impl<'a, K: Ord, V> Entry<'a, K, V> {
 
     /// Ensures a value is in the entry by inserting the result of the default function if empty,
     /// and returns a mutable reference to the value in the entry.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::collections::BTreeMap;
+    ///
+    /// let mut map: BTreeMap<&str, usize> = BTreeMap::new();
+    /// map.entry("poneyland").or_insert_with(|| 12);
```

Comment:
I didn't have any idea of such thing at the moment I wrote the example. I still don't. :-/